lib/web_ui/test/golden_tests/engine/canvas_draw_picture_test.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+import 'dart:html' as html;
+
+import 'package:ui/ui.dart';
+import 'package:ui/src/engine.dart';
+import 'package:test/test.dart';
+
+import 'package:web_engine_tester/golden_tester.dart';
+
+final Rect region = Rect.fromLTWH(0, 0, 500, 100);
+
+void main() async {
+  setUp(() async {
+    debugShowClipLayers = true;
+    SurfaceSceneBuilder.debugForgetFrameScene();
+    for (html.Node scene in html.document.querySelectorAll('flt-scene')) {
+      scene.remove();
+    }
+
+    await webOnlyInitializePlatform();
+    webOnlyFontCollection.debugRegisterTestFonts();
+    await webOnlyFontCollection.ensureFontsLoaded();
+  });
+
+  test('draw growing picture across frames', () async {
+    final SurfaceSceneBuilder builder = SurfaceSceneBuilder();
+    builder.pushClipRect(
+      const Rect.fromLTRB(0, 0, 100, 100),
+    );
+
+    _drawTestPicture(builder, 100, false);
+    builder.pop();
+
+    html.Element elm1 = builder.build().webOnlyRootElement;
+    html.document.body.append(elm1);
+
+    // Now draw picture again but at larger size.
+    final SurfaceSceneBuilder builder2 = SurfaceSceneBuilder();
+    builder2.pushClipRect(
+      const Rect.fromLTRB(0, 0, 100, 100),
+    );
+    _drawTestPicture(builder2, 20, false);
+    builder2.pop();
+
+    elm1.remove();
+    html.document.body.append(builder2.build().webOnlyRootElement);
```

Comment:
This test should also verify that the element was reused. Otherwise, if the canvas does not reuse the image element it will "fix the bug" by virtue of creating a new element instead of reusing.

Reply:
We are reusing the ImageElement, not the canvas. I noticed we don't have such a test for canvas reuse, added one as part of https://github.com/flutter/engine/pull/17378.